services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusyhteystietoEntity.kt

```diff
@@ -60,8 +59,18 @@ class HakemusyhteystietoEntity(
                         puhelin = yhteyshenkilo.hankekayttaja.puhelin,
                         tilaaja = yhteyshenkilo.tilaaja,
                     )
-                }
-        )
+                })
+
+    fun copyToHakemus(hakemus: HakemusEntity) =
```

Comment:
This name implies we're adding the created yhteystieto to the provided hakemus. Instead, this creates a new yhteystieto with the given hakemus in the application field. So maybe `copyWithHakemus()`?

Reply:
The same logic applies to `copyToYhteystieto()`

Reply:
I have some difficulty following the logic in that sentence but you're right that copyWith is better than CopyTo.

Reply:
I tried to say that the method doesn't add to `hakemus.yhteystiedot`, which would be the assumption for a method named `copyToHakemus()`. Instead, it sets `yhteystieto.application`, so `copyWithHakemus()` is more descriptive.